quodlibet/quodlibet/qltk/paned.py

```diff
@@ -119,6 +120,210 @@ class RVPaned(RPaned):
+        """Pass the widget whose pane needs updating."""
+
+        self.__updating = True
+
+        if not isinstance(widget, Gtk.Expander):
+            return
+
+        # lock other non-expander or expanded widgets
+        widgets = self.panelocks(self.root)
+        for w in (w for w in widgets
+            if w is not widget
+                 and (not isinstance(w, Gtk.Expander) or w.get_expanded())):
+            # widget already changed. use stored if available
+            size = w.panelock.size
+#            print_d("%r | setting size request: %d" % (w.id, size))
+            if self.ORIENTATION == Gtk.Orientation.VERTICAL:
+                w.set_size_request(-1, size)
+            else:
+                w.set_size_request(size, -1)
+
+        expanded = widget.get_expanded()
+
+        # set up parent pane(s) position-set
+        parent = self
+        while parent:
+            if isinstance(parent, Gtk.Paned):
+                parent.props.position_set = expanded
+            parent = parent.get_parent()
+
+        if expanded:
+            # widget size changed. use stored if available
+            size = 0
+            try:
+                size = widget.panelock.size
+            except:
+                req = widget.get_requisition()
+                size = req.height if \
+                           self.ORIENTATION == Gtk.Orientation.VERTICAL\
+                           else req.width
+#            print_d("%r | restoring size: %d" % (widget.id, size))
+            if self.ORIENTATION == Gtk.Orientation.VERTICAL:
+                widget.set_size_request(-1, size)
+            else:
+                widget.set_size_request(size, -1)
+        else:
+            # ensure parent pane(s) are not using position-set
+            parent = self
+            while parent:
+                if isinstance(parent, Gtk.Paned):
+                    parent.props.position_set = False
+                parent = parent.get_parent()
+
+            title_size = widget.style_get_property('expander-size') +\
+                         widget.style_get_property('expander-spacing')
+            if widget.get_label_widget():
+                alloc = widget.get_label_widget().get_allocation()
+                if self.ORIENTATION == Gtk.Orientation.VERTICAL:
+                    title_size = max(title_size, alloc.height)
+                else:
+                    title_size = max(title_size, alloc.width)
+            if self.ORIENTATION == Gtk.Orientation.VERTICAL:
+                widget.set_size_request(-1, title_size + 5)
+            else:
+                widget.set_size_request(title_size + 5, -1)
+
+        p = widget
+        while p is not self.root:
+            p = p.get_parent()
+            p.check_resize()
+
+        self.__updating = False
+
+    def do_size_allocate(self, *args):
+        # clear size requests to allow shrinking
+        Gtk.HPaned.do_size_allocate(self, *args)
+
+        if not self.__updating:
```

Comment:
BTW indentation looks neater with a `if self.__updating: return` pattern (as below in fact)

Reply:
yeh, like 10 lines below, that's hilarious (it's actually tragic/painful ..i'm clueless as to why i'd think both ways). changed.